src/mmap_sliver.rs

```diff
@@ -0,0 +1,89 @@
+use std::{ io, slice };
+use std::rc::Rc;
+use std::cell::RefCell;
+
+extern crate libc;
+
+use ::Mmap;
+
+#[cfg(target_os = "windows")]
+use super::windows::MmapInner;
+
+#[cfg(not(target_os = "windows"))]
+use super::posix::MmapInner;
+
+pub struct MmapSliver {
+	// The Rc tracks when to drop
+	// The RefCell allows mutable borrows of parent
+	// for flushing
+	parent: Rc< RefCell<Mmap> >,
+	inner: MmapInner
```

Comment:
I'm about to push a change that exposes `ptr()` and `len()` on `Mmap`, so I think this could be a bit cleaner given that?